docs/configs.md

```diff
@@ -0,0 +1,54 @@
+# Configurations
+
+Madara has multiple custom configurations that change the behavior of some
+Substrate flags, enable settings by default and enable the customization and
+extension for app chains.
+
+## Custom flags
+
+- `--base-path`: Alias to `--madara-path`
+- `--chain-spec-url`: Define an url to retrieve and load a custom chain spec
+- `--configs-url`: Define an url to fetch a Madara configs index file
+- `--dev`: We remapped the dev flag because it was too hard to reproduce the
+  default Substrate dev environment. This flags sets up the following flags:
+  - `--chain dev`
+  - `--force-authoring`
+  - `--alice`
+  - `--rpc-external`
+  - `--rpc-methods unsafe`
+  - We need to enable manually the flag `--rpc-cors all` to accept external
+    connections
+- `--disable-madara-configs`: Disable madara default configs. It won't try to
+  fetch configurations from the cargo project path, nor it will try to fetch
+  from the official repo. With this flag enabled, having a
+  `madara-path/configs/genesis-assets/genesis.json`, it will be enough to run
+  madara
+- `--disable-url-fetch`: Disable the automatic url fetching that we use to
+  retrieve configuration files
+- `--madara-path`: Changes the default [madara-path](#madara-path) (default:
+  `$HOME/.madara`)
+- `--node-key-file`: Key responsible for assigning the peer_id (default:
+  `madara-path/p2p-key.ed25519`)
+- `--testnet`: Allows you to join an official testnet, currently only Sharingan
+  is supported
+- `--update-configs`: By default, if configuration files are already downloaded
+  into `--madara-path`, the client will not try to overwrite them. With this
+  flag, you can force an update
```

Comment:
This will be hard to maintain as the CLI evolve.
Remove it and focus on having a top-notch in-code `clap` documentation, that will appear when you run `.madara --help`.

Reply:
I don't mind deleting it and adding to clap. But it's usual to have people not knowing what flags are available and what is the behavior of the flags we changed (we can't change the clap docs for those). Imo it's better to have the documentation for it

Reply:
The problem is that it will 100% get outdated at some point. While clap `--help` will always be up to date.
And every program in the planet know that he should run `--help` to get the flags of any binary